src/app/sevices/role.service.ts

```diff
@@ -0,0 +1,38 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Observable, take, ReplaySubject, Subject, map } from "rxjs";
+import { Role } from "../models/role-model";
+import { backend_roles } from "../constants/endpoints";
+import { User } from "../models/user";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class RoleService {
+    availableRolesSubject: Subject<Role[]> = new ReplaySubject<Role[]>();
+
+    constructor(private http: HttpClient) {
+        this.getRoles().pipe(take(1)).subscribe(roles => {
```

Comment:
bad practice because this might be called every time you open the website and other users do not need to see the types of roles that the website has.